src/core_plugins/elasticsearch/lib/create_proxy.js

```diff
@@ -20,7 +20,14 @@ function createProxy(server, method, route, config) {
         xforward: true,
         timeout: server.config().get('elasticsearch.requestTimeout'),
         onResponse: function (err, responseFromUpstream, request, reply) {
-          reply(err, responseFromUpstream);
+          const upstreamLocation = responseFromUpstream.headers.location;
```

Comment:
I don't think we can depend on `responseFromUpstream` being defined if there was an error. We should probably early return if there was an error and then read the location header from the `responseFromUpstream `